Mixpanel/Mixpanel.m

```diff
@@ -856,80 +801,23 @@ - (void)archiveProperties
     [p setValue:self.shownSurveyCollections forKey:@"shownSurveyCollections"];
     [p setValue:self.shownNotifications forKey:@"shownNotifications"];
     [p setValue:self.timedEvents forKey:@"timedEvents"];
-    MixpanelDebug(@"%@ archiving properties data to %@: %@", self, filePath, p);
-    if (![NSKeyedArchiver archiveRootObject:p toFile:filePath]) {
-        MixpanelError(@"%@ unable to archive properties data", self);
-    }
-}
-
-- (void)archiveVariants
-{
-    NSString *filePath = [self variantsFilePath];
-    if (![NSKeyedArchiver archiveRootObject:self.variants toFile:filePath]) {
-        MixpanelError(@"%@ unable to archive variants data", self);
-    }
-}
-
-- (void)archiveEventBindings
-{
-    NSString *filePath = [self eventBindingsFilePath];
-    if (![NSKeyedArchiver archiveRootObject:self.eventBindings toFile:filePath]) {
-        MixpanelError(@"%@ unable to archive tracking events data", self);
-    }
+    [self.persistence archiveProperties:p];
+    
+    [self.persistence archiveVariants:self.variants];
+    [self.persistence archiveEventBindings:self.eventBindings];
 }
 
 - (void)unarchive
 {
-    [self unarchiveEvents];
-    [self unarchivePeople];
+    self.eventsQueue = [self.persistence unarchiveEventQueue];
+    self.peopleQueue = [self.persistence unarchivePeopleQueue];
+    
     [self unarchiveProperties];
-    [self unarchiveVariants];
```

Comment:
Oh..so they are done in the `unarchiveProperties`. but why this change? since we have separate logic to handle things like `unarchiveVariants`, do you think it makes more sense to separate them here as well?